presto-orc/src/main/java/com/facebook/presto/orc/writer/FlatMapNumericKeyBlockWriter.java

```diff
@@ -0,0 +1,303 @@
+            MetadataWriter metadataWriter,
+            Type type,
+            OrcType orcKeyType,
+            ColumnWriterFactory valueWriterFactory,
+            List<FlatMapKeyNodeWriter> keyNodes)
+    {
+        this.valueColumnId = valueColumnId;
+        MapType mapType = (MapType) type;
+        this.keyType = mapType.getTypeParameters().get(0);
+        this.valueType = mapType.getTypeParameters().get(1);
+        this.orcKeyType = orcKeyType;
+
+        this.columnWriterOptions = columnWriterOptions;
+        this.dwrfEncryptor = dwrfEncryptor;
+        this.metadataWriter = metadataWriter;
+        this.valueWriterFactory = valueWriterFactory;
+        this.keyNodes = keyNodes;
+
+        this.valueSequenceCounter = 1;
+    }
+
+    public Long getLongValue(Block keysBlock, int position)
+    {
+        Long keyAsLong;
+        SqlFunctionProperties PROPERTIES = SqlFunctionProperties
+                .builder()
+                .setTimeZoneKey(UTC_KEY)
+                .setLegacyTimestamp(true)
+                .setSessionStartTime(0)
+                .setSessionLocale(ENGLISH)
+                .setSessionUser("user")
+                .build();
+
+        switch (orcKeyType.getOrcTypeKind()) {
+            case SHORT:
+                short shortKey = (short) this.keyType.getObjectValue(PROPERTIES, keysBlock, position);
+                keyAsLong = (long) shortKey;
+                break;
+            case INT:
+                int intKey = (int) this.keyType.getObjectValue(PROPERTIES, keysBlock, position);
+                keyAsLong = (long) intKey;
+                break;
+            case LONG:
+                long longKey = (long) this.keyType.getObjectValue(PROPERTIES, keysBlock, position);
+                keyAsLong = longKey;
+                break;
+            default:
+                throw new IllegalStateException("Unexpected value: " + orcKeyType.getOrcTypeKind());
+        }
+        return keyAsLong;
+    }
+
+    @Override
+    public void writeColumnarMap(ColumnarMap columnarMap)
+    {
+        // write keys and value
+        Block keysBlock = columnarMap.getKeysBlock();
+        Block valuesBlock = columnarMap.getValuesBlock();
+
+        for (int position = 0; position < columnarMap.getPositionCount(); position++) {
+            boolean present = !columnarMap.isNull(position);
+            if (!present)
+                continue;
+
+            int offset = columnarMap.getOffset(position);
+            int entryCount = columnarMap.getEntryCount(position);
+            /* This hits array records all the sequence ids 1...n that have a entry for this row */
+            final boolean[] hits = new boolean[keyNodes.size() + 1];
+            for (int entry = 0; entry < entryCount; entry++) {
+                /* Get the key as a Long */
+                Long keyAsLong = getLongValue(keysBlock, offset + entry);
+                /* Get the value for this key as a Block */
```

Comment:
I'd remove it, this comment doesn't really add value because the code is self describing.